test/jest/Netscript/StaticRamParsingCalculation.test.js

```diff
@@ -0,0 +1,500 @@
+      expect(result.importedModules).toHaveLength(1);
+      expect(result.importedModules[0].filePath).toEqual("libTest");
+      expect(result.importedModules[0].alias).toEqual("");
+      expect(result.importedModules[0].imports).toEqual(["dummy", "dummy2"]);
+    });
+
+    it("Import all functions from a library", async function () {
+      const code = `
+        import * as dummy from "libTest";
+        export async function main(ns) {
+          dummy.things();
+        }
+      `;
+      const p = new NetscriptFileParser();
+      const result = p.parseScript(code);
+      expect(result.importedModules).toHaveLength(1);
+      expect(result.importedModules[0].filePath).toEqual("libTest");
+      expect(result.importedModules[0].alias).toEqual("dummy");
+      expect(result.importedModules[0].imports).toEqual(["*"]);
+    });
+
+  });
+
+
+  describe("Building a tree of called functions", function () {
+    it("Find functions inside a single file and their called functions", async function () {
+      const code = `
+        export async function main(ns) {
+          doHacking(ns);
+        }
+        async function doHacking(ns) {
+          await ns.hack("joesguns");
+        }
+      `;
+      const result = new NetscriptFileParser("").parseScript(code);
+      expect(result.functionTree).toHaveLength(2);
+      expect(result.functionTree.map(f => f.fn.name)).toEqual(["main", "doHacking"]);
+      expect(result.functionTree[0].calledFunctions).toEqual([{ name: "doHacking", namespace: "", filePath: ""}]);
+      expect(result.functionTree[1].calledFunctions).toEqual([{ name: "hack", namespace: "ns", filePath: ""}]);
+    });
+
+    it("Find functions and classes inside a single file and their called functions", async function () {
+      const code = `
+        export async function main(ns) {
+          const c = new MyClass();
+          c.doHacking(ns);
+        }
+
+        class MyClass {
+          async doHacking(ns) {
+            await ns.hack("joesguns");
+          }
+        }
+      `;
+      const result = new NetscriptFileParser("").parseScript(code);
+      expect(result.functionTree).toHaveLength(2);
+      expect(result.functionTree.map(f => f.fn.name)).toEqual(["main", "MyClass"]);
+      expect(result.functionTree[0].calledFunctions).toEqual([{ name: "MyClass", namespace: "", filePath: ""}, { name: "doHacking", namespace: "c", filePath: ""}]);
+      expect(result.functionTree[1].calledFunctions).toEqual([{ name: "hack", namespace: "ns", filePath: ""}]);
+    });
+
+    it("Find functions and classes inside a single file and their called functions", async function () {
+      const libCode = `
+        export async function doHack(ns) { return await ns.hack("joesguns"); }
+        export async function doGrow(ns) { return await ns.grow("joesguns"); }
+      `;
+      const lib = new Script(null, "libTest.js", libCode, []);
+
+      const code = `
+        import { doHack } from "libTest";
```

Comment:
dont want to ruin your progress but these needed .js to work correctly
```suggestion
        import { doHack } from "libTest.js";
```

Reply:
(atleast they did in my branch)